src/sentry/api/endpoints/project_stacktrace_links.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import annotations
+
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.integrations.base import IntegrationInstallation
+from sentry.integrations.mixins import RepositoryMixin
+from sentry.models import Project, RepositoryProjectPathConfig
+from sentry.services.hybrid_cloud.integration import integration_service
+from sentry.shared_integrations.exceptions import ApiError
+
+from .project_stacktrace_link import get_code_mapping_configs
+
+MAX_CODE_MAPPINGS_USED = 3
+
+
+class StacktraceLinksSerializer(serializers.Serializer):  # type: ignore
+    file = serializers.ListField(child=serializers.CharField())
+
+    # falls back to the default branch
+    commit_id = serializers.CharField(required=False)
+
+
+@region_silo_endpoint
+class ProjectStacktraceLinksEndpoint(ProjectEndpoint):  # type: ignore
+    """
+    Returns valid links for source code providers so that
+    users can go from the file in the stack trace to the
+    provider of their choice.
+
+    Simular to `ProjectStacktraceLinkEndpoint` but allows
+    for bulk resolution.
+
+    `file`: The file path from the stack trace
```

Comment:
Is this to receive multiple file paths? or just one?

Reply:
multiple, i'll update this comment